pointmatcher/PointCloudGenerator.cpp

```diff
@@ -0,0 +1,360 @@
+    DataPoints pointCloud;
+    addEmpty3dPointFields(numberOfPoints, pointCloud);
+
+    // Create random distribution generators.
+    std::random_device randomDevice;
+    std::mt19937 randomNumberGenerator(randomDevice());
+    std::uniform_real_distribution<ScalarType> uniformDistribution(0.0f, 1.0f);
+
+    // Sampling in Spherical coordinates.
+    auto normalsView{ pointCloud.getDescriptorViewByName("normals") };
+    for (Index i{ 0 }; i < numberOfPoints; ++i)
+    {
+        // Sample random values of theta and phi.
+        const ScalarType theta{ 2.0f * pi * uniformDistribution(randomNumberGenerator) };
+        const ScalarType phi{ std::acos(1.0f - 2.0f * uniformDistribution(randomNumberGenerator)) };
+
+        // Pre-compute values, such as sine and cosine of phi and theta.
+        const ScalarType sinPhi{ std::sin(phi) };
+        const ScalarType cosPhi{ std::cos(phi) };
+        const ScalarType sinTheta{ std::sin(theta) };
+        const ScalarType cosTheta{ std::cos(theta) };
+
+        // Fill features (3D point coordinates)
+        pointCloud.features(0, i) = sinPhi * cosTheta * radius; // x
+        pointCloud.features(1, i) = sinPhi * sinTheta * radius; // y
+        pointCloud.features(2, i) = cosPhi * radius; // z
+
+        // For a sphere, it's possible to compute outward looking normals by simply normalizing the vector going from its center to the
+        // form's body.
+        const StaticCoordVector normalVector{ pointCloud.features.col(i).head(3).normalized() };
+
+        // Fill normals.
+        normalsView(0, i) = normalVector(0);
+        normalsView(1, i) = normalVector(1);
+        normalsView(2, i) = normalVector(2);
+    }
+
+    // Transform point cloud in space.
+    applyTransformation(translation, rotation, pointCloud);
+
+    return pointCloud;
+}
+
+// Reference: https://stackoverflow.com/questions/5837572/generate-a-random-point-within-a-circle-uniformly
+template<typename T>
+typename PointMatcher<T>::DataPoints PointMatcher<T>::PointCloudGenerator::generateUniformlySampledCircle(
+    const ScalarType radius,
+    const Index numberOfPoints,
+    const StaticCoordVector& translation,
+    const Quaternion& rotation)
+{
+    // Create point cloud and add basic structure to fill.
+    DataPoints pointCloud;
+    addEmpty3dPointFields(numberOfPoints, pointCloud);
+
+    // Create random distribution generators.
+    std::random_device randomDevice;
+    std::mt19937 randomNumberGenerator(randomDevice());
+    std::uniform_real_distribution<ScalarType> uniformDistribution(0.0f, 1.0f);
+
+    // Sampling in Cartesian coordinates.
+    auto normalsView{ pointCloud.getDescriptorViewByName("normals") };
+    for (Index i{ 0 }; i < numberOfPoints; ++i)
+    {
+        // Sample random values of theta and phi.
+        const ScalarType phi{ 2.0f * pi * uniformDistribution(randomNumberGenerator) };
+        const ScalarType radiusSample{ sqrt(radius * uniformDistribution(randomNumberGenerator)) };
```

Comment:
I am sorry, but I think you need to take only the square root of the random number, not of the product with the radius. Otherwise, the generated circle will have a radius of sqrt(r).